Sources/Plasma/Apps/plClient/Mac-Cocoa/PLSPatcher.mm

```diff
@@ -0,0 +1,160 @@
+                                       std::placeholders::_1, std::placeholders::_2,
+                                       std::placeholders::_3));
+    _patcher->OnCompletion(std::bind(&Patcher::IOnPatchComplete, _cppPatcher, std::placeholders::_1,
+                                     std::placeholders::_2));
+    _patcher->OnFileDownloadDesired(IApproveDownload);
+    _patcher->OnSelfPatch([&](const plFileName& file) {});
+    _patcher->OnRedistUpdate([&](const plFileName& file) {});
+
+    self.networkPumpTimer = [NSTimer timerWithTimeInterval:1.0 / 1000.0
+                                                   repeats:true
+                                                     block:^(NSTimer* _Nonnull timer) {
+                                                         hsTimer::IncSysSeconds();
+                                                         NetClientUpdate();
+                                                     }];
+
+    return self;
+}
+
+- (void)start
+{
+    [[NSRunLoop mainRunLoop] addTimer:self.networkPumpTimer forMode:NSDefaultRunLoopMode];
+    self.patcher->RequestManifest(plManifest::ClientManifest());
+    self.patcher->Start();
+}
+
+void Patcher::IOnDownloadBegin(const plFileName& file)
+{
+    NSString* fileName = [NSString stringWithSTString:file.AsString()];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [parent.delegate patcher:parent beganDownloadOfFile:fileName];
+    });
+}
+
+void Patcher::IOnProgressTick(uint64_t curBytes, uint64_t totalBytes, const ST::string& status)
+{
+    NSString* statusString = [NSString stringWithSTString:status];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [parent.delegate patcher:parent
+                 updatedProgress:statusString
+                       withBytes:curBytes
+                           outOf:totalBytes];
+    });
+}
+
+bool IApproveDownload(const plFileName& file)
+{
+    ST::string ext = file.GetFileExt();
+    // nothing from Windows, please
+    return ext != "exe" && ext != "pdb" && ext != "dll";
+}
+
+void Patcher::IOnPatchComplete(ENetError result, const ST::string& msg)
+{
+    [parent.networkPumpTimer invalidate];
+    if (IS_NET_SUCCESS(result)) {
+        PLSPatcher* patcher = parent;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [patcher.delegate patcherCompleted:patcher];
+        });
+    } else {
+        NSString* msgString = [NSString stringWithSTString:msg];
+
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [parent.delegate
+                patcherCompletedWithError:parent
+                                    error:[NSError errorWithDomain:@"PLSPatchErrors"
+                                                              code:result
+                                                          userInfo:@{
+                                                              NSLocalizedFailureErrorKey : msgString
+                                                          }]];
+        });
```

Comment:
more odd indentation?

Reply:
Welcome to Obj-C! 😛 

Obj-C indentation says that the `:` should be align in any indented functions inside of each function call. The function calls could be broken up - but this indentation is correct for Obj-C.

This was also partially the basis for if we want to force indentation in Obj-C, or maybe just let things wrap on lines and let the IDE handle things.